File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-2.java
class Solution {
    public int search(int[] nums, int target) {
    int left = 0; 
    int right = nums.length -1; 
    int mid = left + (right-left)/2; 


    while(right > left){ 
        mid = left + (right-left)/2; 
        if(nums[mid] > nums[right]){ 
            left = mid + 1; 
        } else if(nums[mid] < nums[right]){ 
            right = mid; 
        }
    }

    int pivot = left; 
    

    if(target >= nums[pivot] && target <= nums[nums.length -1]) { 
        right =  nums.length-1;
        left = pivot;
    } else { 
        right = pivot -1;
        left = 0; 
    }

     while(right >= left){ 
        mid = left + (right-left)/2; 
        if(nums[mid] == target) {
            return mid;
            }
        if(nums[mid] < target){ 
            left = mid + 1; 
        } else if(nums[mid] > target){ 
            right = mid -1; 
        } 
    }
    return -1;

    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0 
        right = len(numbers)

        while left < right: 
            if(left + right > target): 
                right -= 1
                continue

            if(left + right < target): 
                left += 1 
                continue

    
        return [left + 1, right + 1]

